gitbook-docs/apps_api.md

```diff
@@ -0,0 +1,121 @@
+# Apps API
+
+When a Signal K server provides webapps (rather than just data) it SHOULD implement the Apps API. This provides a standard way to list installed webapps, 
+search for new webapps, and install or remove them. 
+
+This allows different server implementations to share webapps, which gives webapps wider markets, and reduces 'brand' dependency. Basically any 
+Signal K webapp should work on any Signal K server. This API standardises the management of those apps.
+
+To be compatible with any Signal K server Signal K webapps are required to be HTML5 webapps, which rely only on static files served from a web server, 
+and a connection to a Signal K server which supplies json data via the Signal K REST and websocket connections.
+
+## API root
+
+The Aps API root is `signalk/v1/apps/`
+
+## List
+
+`signalk/v1/apps/list` lists the currently install webapps. It is typically used to create app menus, and manage installed app lists.
+It returns a json array in the form:
+```
+[{
+//mandatory
+	"name": "Kip",
+	"description": "Really cool instrument app!!",
+	"location": "../@mxtommy/kip/package/public/index.html",
+	"_location": "../@mxtommy/kip/package/public/index.html", //DEPRECATED
+// if available include manifest.json
+	"manifest": {..content of manifest.json..}
+//recommended
+	"licence": "",
+	"author": "Thomas St.Pierre <thomas@samoht.ca>",
+	"repository": "https://github.com/npm/npm.git",
+	"version": "0.1.8"
+},
+....
+]
+```
+
+## Search
+
+`signalk/v1/apps/search?keyword=signalk-webapp` finds the currently available webapps from npm. It is typically used to list available apps for installation.
+By default it makes a call to `https://api.npms.io/v2/search?size=250&q=keywords:signalk-webapp`
+It returns a json array in the form:
+```
```

Comment:
Shouldn't we include pagination here as part of the search request? I.e. `signalk/v1/apps/search?keyword=foo&skip=0&limit=25`? If things grow that may be necessary

Reply:
Lets add that when we need it. If we add pagination etc it should be generic to all the signalk API's. That includes pagination, ordering,  search hits, max size, lots of stuff etc